### PR TITLE
New article - Data binding - overview

### PR DESCRIPTION
**Pull request by rachana-satao**
This article is merging of three sources: Overview ([link](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/interfaces-related-to-data-binding?view=netframeworkdesktop-4.8), [link](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/data-sources-supported-by-windows-forms?view=netframeworkdesktop-4.8), [link](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/data-binding-and-windows-forms?view=netframeworkdesktop-4.8))

Preview - [link](https://review.docs.microsoft.com/en-us/dotnet/desktop/winforms/data/overview?view=netdesktop-5.0&branch=pr-en-us-1373)

### CODE REVIEW

File: dotnet-desktop-guide/net/winforms/data/overview.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+- Setting the background color of one or more controls.
+
+- Setting the size of controls.
+
+ Essentially, data binding is an automatic way of setting any run-time accessible property of any control on a form.
+
+## Interfaces related to data binding
+
+ADO.NET allows you to create many different data structures to suit the binding needs of your application and the data you're working with. You may want to create your own classes that provide or consume data in Windows Forms. These objects can offer varying levels of functionality and complexity. From basic data binding, to providing design-time support, error checking, change notification, or even support for a structured rollback of the changes made to the data itself.
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/overview.md:35]: lets you create

File: dotnet-desktop-guide/net/winforms/data/overview.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+### Consumers of data binding interfaces
+
+The following sections describe two groups of interface objects. The first group of interface is implemented on data sources by data source authors. The data source consumers such as the Windows Forms controls or components implements these interfaces. The second group of interface is designed to use by component authors. Component authors use these interfaces when they're creating a component that supports data binding to be consumed by the Windows Forms data binding engine. You can implement these interfaces within classes associated with your form to enable data binding. Each case presents a class that implements an interface that enables interaction with data. Visual Studio rapid application development (RAD) data design experience tools already take advantage of this functionality.
+
+#### Interfaces for implementation by data source authors
+
+The Windows Forms controls implements following interfaces:
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/overview.md:43]: Is it controls or control? The sentence doesnt seem right...not sure...

File: dotnet-desktop-guide/net/winforms/data/overview.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+  - "data binding [Windows Forms], interfaces"
+  - "IDataErrorInfo interface [Windows Forms], Windows Forms data binding"
+ms.custom: devdivchpfy22
+---
+
+# Data Binding and Windows Forms
+
+In Windows Forms, you can bind to not just traditional data sources, but also to almost any structure that contains data. You can bind to an array of values that you calculate at run time, read from a file, or derive from the values of other controls.
+
+ In addition, you can bind any property of any control to the data source. In traditional data binding, you typically bind the display property—for example, the <xref:System.Windows.Forms.Control.Text%2A> property of a <xref:System.Windows.Forms.TextBox> control—to the data source. With the .NET, you also have the option of setting other properties through binding as well. You might use binding to perform the following tasks:
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/overview.md:23]: Is there an additional space here?

Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/overview.md:23]: With the .NET ---> Can we just say With .NET?

Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/overview.md:23]: With the .NET, you also have the option of setting other properties through binding as well. --> Remove as well, you have already used "also" in the sentence earlier.

File: dotnet-desktop-guide/net/winforms/data/overview.md (hunk 1 above)
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/overview.md:35]: You may want --> You might

File: dotnet-desktop-guide/net/winforms/data/overview.md [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+  The type of change is described by the <xref:System.ComponentModel.ListChangedType> property of the <xref:System.ComponentModel.ListChangedEventArgs> parameter. Hence, whenever the data model is updated, any dependent views, such as other controls bound to the same data source, will also be updated. However, objects contained within the list will have to notify the list when they change so that the list can raise the <xref:System.ComponentModel.IBindingList.ListChanged> event.
+
+  > [!NOTE]
+  > The <xref:System.ComponentModel.BindingList%601> provides a generic implementation of the <xref:System.ComponentModel.IBindingList> interface.
+
+- <xref:System.ComponentModel.IBindingListView> interface
+
+  A class that implements the <xref:System.ComponentModel.IBindingListView> interface provides all the functionality of an implementation of <xref:System.ComponentModel.IBindingList>, along with filtering and advanced sorting functionality. This implementation offers string-based filtering, and multi column sorting with property descriptor-direction pairs.
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/overview.md:66]: multi-column

File: dotnet-desktop-guide/net/winforms/data/overview.md [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+|Data entry|A common way to enter large amounts of related data or to prompt users for information is through a data entry form. Users can enter information or select choices using text boxes, option buttons, drop-down lists, and check boxes. Information is then submitted and stored in a database, whose structure is based on the information entered.|
+|Master/detail relationship|A master/detail application is one format for looking at related data. Specifically, there are two tables of data with a relation connecting them—in the classic business example, a "Customers" table and an "Orders" table with a relationship between them linking customers and their respective orders. For more information about creating a master/detail application with two Windows Forms <xref:System.Windows.Forms.DataGridView> controls, see [How to: Create a Master/Detail Form Using Two Windows Forms DataGridView Controls](/dotnet/desktop/winforms/controls/create-a-master-detail-form-using-two-datagridviews?view=netframeworkdesktop-4.8&preserve-view=true)|
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/overview.md:240]: why is there a hyphen between them-in?

File: dotnet-desktop-guide/net/winforms/data/overview.md [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+#### Interfaces for implementation by component authors
+
+The following interfaces are designed for consumption by the Windows Forms data-binding engine:
+
+- <xref:System.Windows.Forms.IBindableComponent> interface
+
+  A class that implements this interface is a non-control component that supports data binding. This class returns the data bindings and binding context of the component through the <xref:System.Windows.Forms.IBindableComponent.DataBindings%2A> and <xref:System.Windows.Forms.IBindableComponent.BindingContext%2A> properties of this interface.
+
+  > [!NOTE]
+  > If your component inherits from <xref:System.Windows.Forms.Control>, you do not need to implement the <xref:System.Windows.Forms.IBindableComponent> interface.
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/overview.md:158]: do not --> don't

File: dotnet-desktop-guide/net/winforms/data/overview.md (hunk 3 above)
Comment (rachana-satao) [reply dotnet-desktop-guide/net/winforms/data/overview.md:23]: no

File: dotnet-desktop-guide/net/winforms/data/overview.md [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+
+### Consumers of data binding interfaces
+
+The following sections describe two groups of interface objects. The first group of interface is implemented on data sources by data source authors. The data source consumers such as the Windows Forms controls or components implements these interfaces. The second group of interface is designed to use by component authors. Component authors use these interfaces when they're creating a component that supports data binding to be consumed by the Windows Forms data binding engine. You can implement these interfaces within classes associated with your form to enable data binding. Each case presents a class that implements an interface that enables interaction with data. Visual Studio rapid application development (RAD) data design experience tools already take advantage of this functionality.
```
[the remark this answers is not in the document]
Comment (rachana-satao) [reply dotnet-desktop-guide/net/winforms/data/overview.md:39]: implement

**Comment by rachana-satao** [comment]
@adegeo Acrolinx score for the article - 99. Kind request to review PR - https://github.com/dotnet/docs-desktop/pull/1373

File: dotnet-desktop-guide/net/winforms/data/overview.md [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+  - "IList interface [Windows Forms], Windows Forms data binding"
+  - "IBindingList interface [Windows Forms], Windows Forms data binding"
+  - "interfaces [Windows Forms], Windows Forms data binding"
+  - "IEditableObject interface [Windows Forms], Windows Forms data binding"
+  - "data binding [Windows Forms], interfaces"
+  - "IDataErrorInfo interface [Windows Forms], Windows Forms data binding"
+ms.custom: devdivchpfy22
+---
+
+# Data Binding and Windows Forms
```
Comment (adegeo) [inline dotnet-desktop-guide/net/winforms/data/overview.md:19]:
```suggestion
# Data Binding (Windows Forms .NET)
```

File: dotnet-desktop-guide/net/winforms/data/overview.md [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+
+- <xref:System.ComponentModel.IEditableObject> interface
+
+  A class that implements the <xref:System.ComponentModel.IEditableObject> interface allows an object to control when changes to that object are made permanent. This implementation supports the <xref:System.ComponentModel.IEditableObject.BeginEdit%2A>, <xref:System.ComponentModel.IEditableObject.EndEdit%2A>, and <xref:System.ComponentModel.IEditableObject.CancelEdit%2A> methods, which enable you to roll back changes made to the object. Following is a brief explanation of the functioning of the <xref:System.ComponentModel.IEditableObject.BeginEdit%2A>, <xref:System.ComponentModel.IEditableObject.EndEdit%2A>, and <xref:System.ComponentModel.IEditableObject.CancelEdit%2A> methods and how they work with one another to enable a possible rollback of changes made to the data:
```
Comment (adegeo) [inline dotnet-desktop-guide/net/winforms/data/overview.md:70]: Can you clean up all of the xref links in the article paragraphs? There are just too many and it's a bit distracting to read. all of the old windows forms content is like this, yuck!

Within the same paragraph once the first XREF of a type has been used, just change the next mention to `inline code`. For example in this paragraph:

```suggestion
  A class that implements the <xref:System.ComponentModel.IEditableObject> interface allows an object to control when changes to that object are made permanent. This implementation supports the <xref:System.ComponentModel.IEditableObject.BeginEdit%2A>, <xref:System.ComponentModel.IEditableObject.EndEdit%2A>, and <xref:System.ComponentModel.IEditableObject.CancelEdit%2A> methods, which enable you to roll back changes made to the object. Following is a brief explanation of the functioning of the `BeginEdit`, `EndEdit`, and `CancelEdit` methods and how they work with one another to enable a possible rollback of changes made to the data:
```

The next paragraph is even worse. This is how it looks rendered; just too many links

image

**Review by adegeo — CHANGES_REQUESTED** [review]
Thanks! Here are some things to consider.

File: dotnet-desktop-guide/net/winforms/data/overview.md (hunk 9 above)
Comment (rachana-satao) [reply dotnet-desktop-guide/net/winforms/data/overview.md:70]: Ok

**Review by adegeo — APPROVED** [review]

**Pull request merged**